aiida/backends/sqlalchemy/migrations/versions/12536798d4d3_trajectory_symbols_to_attribute.py

```diff
@@ -33,24 +33,36 @@
 
 def upgrade():
     """Migrations for the upgrade."""
-    session = Session(bind=op.get_bind())
-    trajectories = session.query(DbNode).filter_by(type='node.data.array.trajectory.TrajectoryData.').all()
-    for t in trajectories:
-        symbols = load_node(pk=t.id).get_array('symbols').tolist()
-        t.attributes['symbols'] = symbols
-        flag_modified(t, 'attributes')
-        session.add(t)
-    session.commit()
-    session.close()
+    # yapf:disable
+    connection = op.get_bind()
+
+    DbNode = table('db_dbnode', column('id', Integer), column('uuid', UUID), column('type', String),
+                   column('attributes', JSONB))
+
+    nodes = connection.execute(
+        select([DbNode.c.id, DbNode.c.uuid]).where(
+            DbNode.c.type == op.inline_literal('node.data.array.trajectory.TrajectoryData.'))).fetchall()
+
+    for pk, uuid in nodes:
+        symbols = load_numpy_array_from_repository(uuid, 'symbols').tolist()
+        connection.execute(DbNode.update().where(DbNode.c.id == pk).values(
+            attributes=func.jsonb_set(DbNode.c.attributes, op.inline_literal('{"symbols"}'), cast(symbols, JSONB))))
 
 
 def downgrade():
     """Migrations for the downgrade."""
-    session = Session(bind=op.get_bind())
-    trajectories = session.query(DbNode).filter_by(type='node.data.array.trajectory.TrajectoryData.').all()
-    for t in trajectories:
-        t.delete_attribute('symbols')
-        flag_modified(t, 'attributes')
-        session.add(t)
-    session.commit()
-    session.close()
+    # yapf:disable
+    connection = op.get_bind()
+
+    DbNode = table('db_dbnode', column('id', Integer), column('uuid', UUID), column('type', String),
+                   column('attributes', JSONB))
+
+    nodes = connection.execute(
+        select([DbNode.c.id, DbNode.c.uuid]).where(
+            DbNode.c.type == op.inline_literal('node.data.array.trajectory.TrajectoryData.'))).fetchall()
+
+    for pk, _ in nodes:
+        connection.execute(
```

Comment:
Good catch, in fact it seems like the second one in SqlAlchemy should not work because the `json_object_delete_keys` function does not even exist. This shows a problem with our testing: this was not caught because we do not tests the downgrades explicitly. Sure it is called when downgrading, but only during empty databases mostly, which is why the faulty code was not triggered since it is only ran when there are TrajectoryData nodes.